version: "3.9"
services:
  user-service:
    image: ${BACKEND_IMAGE:-ghcr.io/kotaorie/learningai/backend:staging}
    ports:
      - "3001:3001"
    depends_on:
      postgres-db:
        condition: service_healthy
    environment:
      DATABASE_URL: "postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@postgres-db:5432/${POSTGRES_DB}"
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      POSTGRES_DB: ${POSTGRES_DB}
      DB_HOST: postgres-db
      DB_PORT: 5432
    networks:
      - app_network
    restart: unless-stopped

  nuxt-app:
    image: ${FRONTEND_IMAGE:-ghcr.io/kotaorie/learningai/frontend:staging}
    ports:
      - "3000:3000"
    networks:
      - app_network
    restart: unless-stopped
    depends_on:
      - user-service

  postgres-db:
    image: postgres:15-alpine
    container_name: postgres_db
    environment:
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      POSTGRES_DB: ${POSTGRES_DB}
    ports:
      - "5432:5432"
    volumes:
      - postgres_data:/var/lib/postgresql/data
      - ./back/db/init.sql:/docker-entrypoint-initdb.d/init.sql:ro
    networks:
      - app_network
    restart: unless-stopped
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${POSTGRES_USER} -d ${POSTGRES_DB}"]
      interval: 30s
      timeout: 10s
      retries: 5

volumes:
  postgres_data:
    driver: local

networks:
  app_network:
    driver: bridge